### PR TITLE
PUBDEV-7269 Multinomial AUC/AUCPR

### PR DESCRIPTION
**Pull request by maurever**
# Multinomial AUC/PR AUC

Todo:
- [x]  implement OVR and OVO calculation 
- [x] implement macro and weighted average
- [x] return all metrics in table
- [x] Python API
- [x] R API
- [x] Java test (it returns correct values, grid search test, early stopping test)
- [x] Python, R tests
- [x] add parameter to set default `auc()` and `prauc()` value (for grid search and early stopping)
- [x] set AUC/PR AUC in grid search for multinomial metric
- [x] set AUC/PR AUC in early stopping for multinomial metric
- [x] solve metric construction in GenericModel

- [x] demo
- [x] test performance
- [x] update documentation AUC/PR AUC page
- [x] decide the API is designed correctly 
- [x] decide what is the limit of domain size
- [x] disable by default

### 2. User can get whole multinomial AUC/PR AUC summary using `multinomial_auc_table()` and  `multinomial_aucpr_table()` method.

api

### 3. Users can change the default method to calculate AUC/PR AUC using `auc_type` parameter in the model. The possibilities are  `NONE`, `MACRO_OVR`, `WEIGHTED_OVR`, `MACRO_OVO`, `WEIGHTED_OVO`,  `AUTO`.

auc_type

### 4. The limit of domain to print this summary in H2O logs is 20 (same as for confusion matrix). The limit to calculate multinomial AUC/PR AUC is currently 50.

too_big

### 6. User can use `AUC` and `AUCPR` metric for early stopping in multinomial problems.

early_stopping

### 7. User can get `auc` and `aucpr` metric in grid search and can sort result models using this metric.

gridsearch

## Performance

100 trees **without** calculating AUC/PR AUC: **0 min 30.805 sec (~ 31 sec)**
100 trees **with** calculating      AUC/PR AUC: **4 min 20.068 sec (~264 sec)**

performance

JIRA: https://0xdata.atlassian.net/jira/software/c/projects/PUBDEV/issues/PUBDEV-7269

### CODE REVIEW

**Review by neemah2o — APPROVED** [review]
LGTM

File: h2o-core/src/test/java/water/rapids/ast/prims/advmath/AstUniqueTest.java [hunk 1]
```diff
@@ -124,5 +125,30 @@ public void categoricalNasRecognized() {
     }
   }
 
+  @Test
+  public void stringsUnsupported() {
```
Comment (michalkurka) [inline h2o-core/src/test/java/water/rapids/ast/prims/advmath/AstUniqueTest.java:129]: This looks like a wrong merge. This code is already in master.

**Comment by michalkurka** [comment]
Please make sure you rebase this branch - currently it looks like you are changing files that this PR shouldn't be changing - eg. TransformWrappedVec.java

**Comment by michalkurka** [comment]
Otherwise it IMHO looks good -thank you for this contribution!

**Comment by maurever** [comment]
> Please make sure you rebase this branch - currently it looks like you are changing files that this PR shouldn't be changing - eg. TransformWrappedVec.java

Ou no, you are true! I rebased on rel-zermelo but this PR is on master. Fixed.

**Review by michalkurka — APPROVED** [review]
LGTM!!!

**Comment by maurever** [comment]
Fix AutoML leaderboard and tests, the failed tests look unrelated and are on the master branch too. So I am merging.

**Pull request merged**